addons/account/migrations/9.0.1.1/pre-migration.py

```diff
@@ -370,6 +370,38 @@ def set_date_maturity(env):
     )
 
 
+def migrate_bic_numbers(env):
+    """ BIC numbers migrated in supplier bank details """
+    # if no bank exists with a matching BIC, automatically create a new bank, with the new BIC;
+    # if the bank account details do not specify the bank name, use the BIC as bank name.
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO res_bank(name, bic)
+        SELECT CASE WHEN bank_name IS NOT NULL THEN bank_name ELSE bank_bic END, bank_bic
+        FROM res_partner_bank
+        WHERE bank IS NULL AND bank_bic IS NOT NULL AND bank_bic NOT IN (
+            SELECT bic
+            FROM res_bank
+            WHERE bic IS NOT NULL
+        )
+        GROUP BY bank_name, bank_bic
```

Comment:
Even if the banks have a different name, I would advice against introducing multiple banks with the same BIC